### DISCUSSION TITLE
When to use an uncontrolled vs. controlled component?

### DISCUSSION THREAD

**Discussion by tordans** [uber/react-map-gl#2307] [Q&A]
I only just realized there is a page https://visgl.github.io/react-map-gl/docs/get-started/state-management just on the topic of controlled vs. uncontrolled components. I have been using uncontrolled components all the time … and I wonder, when would I even want to use controlled components?
What are the advantages of the one vs. the other?

**Comment by KiwiKilian** [comment]
Best example is possibly the [side-by-side view](https://visgl.github.io/react-map-gl/examples/side-by-side). So basically it's helpful if you want to process the current view state of the map or change it programatically.
